services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaController.kt

```diff
@@ -232,6 +233,54 @@ Removes the token after a successful identification.
         hankeKayttajaService.createPermissionFromToken(currentUserId(), tunnistautuminen.tunniste)
     }
 
+    @PostMapping("/kayttajat/{kayttajaId}/kutsu")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @Operation(
+        summary = "Resend an invitation email",
+        description =
+            """
+Resend the invitation email the user was sent when they were first added to the
+hanke as a contact person.
+
+Regenerates the invitation token and link. This means that the link in the
+original email will not work anymore. It also means that the period of validity
+of the token and link will be reset.
+"""
+    )
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(
+                    description = "Invitation email resent",
+                    responseCode = "204",
+                ),
+                ApiResponse(
+                    description = "User not found",
+                    responseCode = "404",
+                ),
+                ApiResponse(
+                    description =
+                        "User has already been activated or the current user doesn't have name and email registered.",
+                    responseCode = "409",
+                    content = [Content(schema = Schema(implementation = HankeError::class))]
+                ),
+            ]
+    )
+    fun resendInvitations(@PathVariable kayttajaId: UUID) {
+        featureFlags.ensureEnabled(Feature.USER_MANAGEMENT)
+
+        val kayttaja =
+            hankeKayttajaService.getKayttajaById(kayttajaId)
+                ?: throw HankeKayttajaNotFoundException(kayttajaId)
+        permissionService.hasPermission(
```

Comment:
Should this use the new authorisation?

Reply:
Rebased this and changed the authorization to the new style.